### PR TITLE
Fixed borgs and AI shells opening doors while inactive

### LINKED ISSUE #6868: Unintended and rather cheap way to open restricted doors
### Labels

[BUG]

### Description

Pushing any, including uncontrolled, borg/AI shell/possibly more, into a door will open it with seemingly almost no restrictions. The armory and Head of Security office would not open this way, however other doors in security and other areas would including AI upload.

### Steps to reproduce

1. Grab an AI shell/borg
2. Drag it in front of a door
3. Give a good hard push

### Expected Behavior

The door should not open if it is restricted. AI shells/borgs wouldn't open the door simply by being near it but through remote control. 

### Additional Information & Screenshots

I spoke to an admin about it and they advised I should probably report it. It might not be a "bug" but more an overlooked consequence, however an antagonist could easily take advantage of an unused AI shell and figured for balance sake it probably shouldn't work the way it does currently.

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[BUG]

## About the PR 
Fixes #6868
Just checks if a silicon mob has a mind before letting it bump open a door. Not sure if this would break NPC silicons, I don't know enough about NPC code. Also not sure if this is a bugfix or a balance change, so feel free to re-flair it.

## Why's this needed? 
AI shells being left around important areas while inactive is extremely common and they shouldn't act as a draggable AA ID.

### CODE REVIEW

**Comment by ThatFiveGuys** [comment]
I enjoy things like being able to use an spontaneous unit borg to break into places, only for a player to join the game as borg and catch you in the act. Would not at all be against removing access from the brains who cant be player controlled at all such as monkey brains, however unsure how well this could be conveyed.

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/obj/machinery/door/airlock.dm [hunk 1]
```diff
@@ -1311,8 +1311,12 @@ About the new airlock wires panel:
 		if(world.time - src.last_bump <= 30)
 			return
 
-		if (issilicon(AM) && (aiControlDisabled == 1 || cant_emag))
-			return
+		if (issilicon(AM))
+			if (aiControlDisabled == 1 || cant_emag)
+				return
+			var/mob/silicon = AM
+			if (isnull(silicon.mind))
```
[the remark this answers is not in the document]
Comment (TobleroneSwordfish) [reply code/obj/machinery/door/airlock.dm:1318]: gotcha

**Comment by TobleroneSwordfish** [comment]
Personally I just don't like the idea of having to keep a constant eye on my AI shells in case some staffie is using them to break into sec. Not to mention that being somewhere with an AI shell looks a lot less suspicious than hacking and it's easy for people to assume the AI let them in for a good reason and then dropped the shell to respond to something else.

Obviously it's a balance question so maintainers have final say in any case.

**Review by virvatuli-gh — COMMENTED** [review]
I was the one who asked the bug-reporter to report it, because it's not intentional as far as I know and personally I've always hated it. It's a borderline exploit that nobody has bothered to fix and people just accepted as a "feature", despite it not making any sense.  Thanks for taking the initiative to fix it!

**Review by pali6 — APPROVED** [review]

**Pull request merged**